src/App.jsx

```diff
@@ -14,33 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [activeSort, setActiveSort] = useState(null);
+  const [isReversed, setIsReversed] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const isOriginalOrder = () => {
+    return JSON.stringify(goods) === JSON.stringify(goodsFromServer);
+  };
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const sortAlphabetically = () => {
+    const sorted = [...goods].sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
+    setGoods(isReversed ? sorted.reverse() : sorted);
+    setActiveSort('alphabetically');
+  };
+
+  const sortByLength = () => {
+    const sorted = [...goods].sort((a, b) => {
+      const lengthDiff = a.length - b.length;
+
+      if (lengthDiff === 0) {
+        return a.localeCompare(b);
+      }
+
+      return lengthDiff;
+    });
+
+    setGoods(isReversed ? sorted.reverse() : sorted);
+    setActiveSort('length');
+  };
+
+  const reverse = () => {
+    setGoods([...goods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  const reset = () => {
+    setGoods(goodsFromServer);
+    setActiveSort(null);
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${activeSort !== 'alphabetically' ? 'is-light' : ''}`}
```

Comment:
To set classes conditionally it's better to use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library